### PR TITLE
fix(Go): Update array_queue.go and array_deque.go

### PR DESCRIPTION
**Pull request by XiaoK29**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: codes/go/go.mod [hunk 1]
```diff
@@ -1,3 +1,11 @@
 module github.com/krahets/hello-algo
 
 go 1.19
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
```
Comment (krahets) [inline codes/go/go.mod:6]: Are these libs necessary?

File: codes/go/go.sum [hunk 2]
```diff
@@ -0,0 +1,10 @@
+github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
```
Comment (krahets) [inline codes/go/go.sum:1]: Same as above

**Review by krahets — COMMENTED** [review]
Thanks for the PR! Please adress the comments

File: codes/go/go.mod (hunk 1 above)
Comment (XiaoK29) [reply codes/go/go.mod:6]: > Are these libs necessary?

The testify depends on this package go-spew

Comment (krahets) [reply codes/go/go.mod:6]: We'd like to keep the third-party libraries as simple as possible, so I'd recommend remove them and the corresponding testing code.

Comment (XiaoK29) [reply codes/go/go.mod:6]: > We'd like to keep the third-party libraries as simple as possible, so I'd recommend remove them and the corresponding testing code.

好的

**Review by krahets — APPROVED** [review]

**Pull request merged**